[Fix #6597] - LineEndConcatenation is unsafe

jaredbeck opened the pull request:
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

koic commented on config/default.yml:
```diff
@@ -3352,6 +3352,7 @@ Style/LineEndConcatenation:
                  Use \ instead of + or << to concatenate two string literals at
                  line end.
   Enabled: true
+  Safe: false
```
Maybe I think that `SafeAutoCorrect` rather than `Safe` should be specified.

```diff
-Safe: false
+SafeAutoCorrect: false
```

koic commented on CHANGELOG.md:
```diff
@@ -5,6 +5,7 @@
 ### New features
 
 * [#6580](https://github.com/rubocop-hq/rubocop/pull/6580): New cop `Rails/LinkToBlank` checks for `link_to` calls with `target: '_blank'` and no `rel: 'noopener'`. ([@Intrepidd][])
+* [#6597](https://github.com/rubocop-hq/rubocop/issues/6597): `Style/LineEndConcatenation` is now known to be un-`Safe`. ([@jaredbeck][])
```
It's a change in behavior so write it in the `### Changes` section.

jaredbeck commented:
@koic changes made, thanks!

Drenmi commented:
@jaredbeck This change is good to go. Just needs a rebase and regenerating of the documentation. 🙇

koic commented on config/default.yml:
```diff
@@ -3352,7 +3352,9 @@ Style/LineEndConcatenation:
   Enabled: true
+  SafeAutoCorrect: false
   VersionAdded: '0.18'
+  VersionChanged: '0.63'
```
I'm sorry I'm late about merging 💦
Please update this to `'0.64'`,  regenerate the document (Run `rake generate_cops_documentation` command) and solve CHANGELOG.md with the latest master branch.

jaredbeck commented:
> I'm sorry I'm late about merging 💦 Please update this to '0.64', regenerate the document (Run rake generate_cops_documentation command) and solve CHANGELOG.md with the latest master branch.

Rebased and updated as requested, thanks Ito-san.
